R/RLearner_classif_xgboost.R

```diff
@@ -79,6 +80,10 @@ trainLearner.classif.xgboost = function(.learner, .task, .subset, .weights = NUL
 
   task.data = getTaskData(.task, .subset, target.extra = TRUE)
   label = match(as.character(task.data$target), td$class.levels) - 1
+
+  # recode to 0:1 to that for the binary case the positive class translates to 1 (https://github.com/mlr-org/mlr3learners/issues/32)
+  # task.data$target is guaranteed to have the factor levels in the right order
```

Comment:
have you doubled checked that this holds for mlr?

Reply:
Yes. Also the tests should ensure this.